### PR TITLE
Get BiosCfg ready for prod

### PR DESCRIPTION
**Pull request by jakeschuurmans**
#### What does this PR do

Attempted to put the work in their own commits to help reading, and so I dont need as many PRs.

- Add github runner to package helm chart, making it available at https://metal-toolbox.github.io/bioscfg
- Update Helm Chart to get it ready for the sandbox and prod
- Update docker to latest alpine
- Fixes for the makefile
- Refactor CMD, getting it ready for future work
- Replace oldTask and steps with rivets.Task
  - Going with a simpler approach, instead of these complicated pieces in steps.go, just a simple function that does the work. Easier to read, easier to add new actions and steps.
- Replace slog with logrus. I know we want to move off logrus, but I think we need to do this at a later date.
  - Before using slog, I want to make sure it does exactly what we want it to.
- Move handler.go to bioscfg, plus a bit of a refactor to match changes done to CMD
- Move publisher to bioscfg, in its own file, it shouldnt be tagged onto tasks.go

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
-FROM alpine:3.8 as runner
+FROM alpine:latest
+
+ENTRYPOINT ["/usr/sbin/bioscfg"]
```
Comment (coffeefreak101) [inline Dockerfile:3]: Any reason why we moved the `entrypoint` up here? I don't know that it will affect the execution, but I've never really seen it NOT be the last line of Dockerfile.

File: internal/bioscfg/bioscfg.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+)
+
+var (
+	pkgName = "internal/bioscfg"
+)
+
+// BiosCfg BiosCfg Controller Struct
+type BiosCfg struct {
+	cfg     *config.Configuration
+	logger  *logrus.Entry
```
Comment (coffeefreak101) [inline internal/bioscfg/bioscfg.go:21]: I thought we were moving away from `logrus`?

Also, I really don't like having to pass around the logger. It really limits your ability to log, unless you include the logger everywhere, which then bloats the code because you have to have the logger as a field on every struct and include it in every `new()` method.

File: internal/store/fleetdb/client.go [hunk 3]
```diff
@@ -89,13 +80,13 @@ func newFleetDBClientWithOAuthOtel(ctx context.Context, cfg *configuration.Fleet
...
-	provider, err := oidc.NewProvider(ctx, cfg.OidcIssuerEndpoint)
+	provider, err := oidc.NewProvider(ctx, cfg.OidcIssuerURL)
 	if err != nil {
 		return nil, err
 	}
 
-	// clientID defaults to 'bioscfg'
-	clientID := "bioscfg"
+	// clientID defaults to 'flipflop'
+	clientID := "flipflop"
```
Comment (coffeefreak101) [inline internal/store/fleetdb/client.go:89]: Why was this changed to flipflop, or was this copy pasta?

File: internal/bioscfg/bioscfg.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+	fleetdb *fleetdb.Store
+	nc      *ctrl.NatsController
+}
+
+// New create a new BiosCfg Controller
+func New(ctx context.Context, cfg *config.Configuration, logger *logrus.Entry) (*BiosCfg, error) {
+	bc := &BiosCfg{
+		cfg:    cfg,
+		logger: logger,
+		ctx:    ctx,
```
Comment (coffeefreak101) [inline internal/bioscfg/bioscfg.go:32]: Contexts are not meant to be stored in a struct:

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it.

https://pkg.go.dev/context

There's probably nothing bad that will happen if we do add it to a struct, but best to use them as intended.

File: internal/bioscfg/bioscfg.go (hunk 2 above)
Comment (jakeschuurmans) [reply internal/bioscfg/bioscfg.go:21]: I vageuly remember us talking about it, but we havent really done it anywhere else. I am 100% down to remove thirdparty dependencies and move to slog. I will take a look.

File: Dockerfile (hunk 1 above)
Comment (jakeschuurmans) [reply Dockerfile:3]: no, ill move it back. I forgot to change it back. I was dealing with a weird bug and trying some wild stuff to see what was up.

File: internal/bioscfg/bioscfg.go (hunk 2 above)
Comment (jakeschuurmans) [reply internal/bioscfg/bioscfg.go:21]: After talking with John, we are deciding to postpone the work to move over to logrus until a later date

**Review by coffeefreak101 — APPROVED** [review]
